src/pint/models/model_builder.py

```diff
@@ -24,295 +25,402 @@
+        """
+        all_systems = self.category_component_map['pulsar_system']
+        result = None
+        # Search the system name first
+        print(system_name)
+        if system_name in all_systems:
+            result = self.components[system_name]
+        else: # search for the pulsar system aliases
+            for cp_name in all_systems:
+                if system_name == self.components[cp_name].binary_model_name:
+                    result = self.components[cp_name]
                 else:
-                    if cpi.is_in_parfile(params_inpar):
-                        selected_c = cpi
-            if selected_c is not None:
-                self.select_comp[cat] = selected_c
+                    continue
+        if result is None:
+            raise UnknownBinaryModel("Pulsar system/Binary model component {}"
+                                     " is not provided.".format(system_name))
+        else:
+            return result
 
-    def sort_components(self, category_order=DEFAULT_ORDER):
-        """Sort the components into order.
+    def alias_2_pint_param(self, alias):
+        """ Translate the alias to a PINT parameter name.
+        """
+        pint_par = self.param_alias_map.get(alias, None)
+        # If it is not in the map, double check if it is a repeatable par.
+        if pint_par is None:
+            try:
+                prefix, idx_str, idx = split_prefixed_name(alias)
+                # assume the index 1 parameter is in the alias map
+                # count length of idx_str and dectect leading zeros
+                num_lzero = len(idx_str) - len(str(idx))
+                if num_lzero > 0: # Has leading zero
+                    fmt = len(idx_str)
+                else:
+                    fmt = 0
+                # Handle the case of start index from 0 and 1
+                for start_idx in [0, 1]:
+                    example_name = prefix + '{1:0{0}}'.format(fmt, start_idx)
+                    pint_par = self.param_alias_map.get(example_name, None)
+                    if pint_par:
+                        break
+                if pint_par: # Find the start parameter index
+                    pint_par = split_prefixed_name(pint_par)[0] + idx_str
+            except PrefixError:
+                pint_par = None
+        return pint_par
+
+    def _get_param_overlap_rank(self, components, params, primary_key):
+        """ Get the rank of how components match a given set of paramters
 
         Parameters
         ----------
-        category_order: list, optional
-           The order for the order sensitive component categories.
+        components: list
+            A list of the component names.
+        params: list
+            A list of matching parameter names.
+        primary_key: str
+            The primary compare key(e.g., `params` or `component_special_params`)
+
+        Return
+        ------
+        A sorted tuple list based on how many overlap parameters.
```

Comment:
I do not think that this is an intelligible criterion for users. Requiring them to count how many parameters match different models to know how their par file will be interpreted is not acceptable. This does not appear to ever be used anyway, so please remove it.